xdsl/tools/tblgen_to_py.py

```diff
@@ -0,0 +1,566 @@
+import argparse
+import json
+import subprocess
+import textwrap
+from dataclasses import dataclass, field
+from enum import Enum
+from io import StringIO
+from keyword import iskeyword
+from sys import stdin
+from typing import Any
+
+
+@dataclass
+class TblgenRecord:
+    js: Any
+
+    def __getitem__(self, i: str) -> Any:
+        return self.js[i]
+
+    @property
+    def name(self) -> str:
+        return self["!name"]
+
+    @property
+    def summary(self) -> str:
+        return self["summary"]
+
+    @property
+    def superclasses(self) -> set[str]:
+        return set(self["!superclasses"])
+
+    @property
+    def dialect(self) -> str:
+        return self["dialect"]["def"]
+
+
+class TblgenOp(TblgenRecord):
+    @property
+    def dialect(self) -> str:
+        return self["opDialect"]["def"]
+
+    @property
+    def opName(self) -> str:
+        return self["opName"]
+
+    @property
+    def assemblyFormat(self) -> str | None:
```

Comment:
```suggestion
    def assembly_format(self) -> str | None:
```
And same things on other camelCase functions to make them snake_case (to follow Python style).